src/state/apis/swappers/swappersApi.ts

```diff
@@ -30,83 +51,57 @@ export const swappersApi = createApi({
-          } catch (error) {
-            console.error(error)
-          }
-        if (ThorSwap)
-          try {
-            const quote: Result<TradeQuote2, SwapErrorRight> = await getThorTradeQuoteHelper(
-              ...quoteHelperArgs,
-            )
-            quotes.push({
-              ...quote,
-              swapperName: SwapperName.Thorchain,
-            })
-          } catch (error) {
-            console.error(error)
-          }
-        if (ZrxSwap)
-          try {
-            const quote: Result<TradeQuote2, SwapErrorRight> = await getZrxTradeQuoteHelper(
-              ...quoteHelperArgs,
-            )
-            quotes.push({
-              ...quote,
-              swapperName: SwapperName.Zrx,
-            })
-          } catch (error) {
-            console.error(error)
-          }
-        if (OneInch)
-          try {
-            const quote: Result<TradeQuote2, SwapErrorRight> = await getOneInchTradeQuoteHelper(
-              ...quoteHelperArgs,
-            )
-            quotes.push({
-              ...quote,
-              swapperName: SwapperName.OneInch,
-            })
-          } catch (error) {
-            console.error(error)
-          }
-        if (Cowswap)
-          try {
-            const quote: Result<TradeQuote2, SwapErrorRight> = await getCowTradeQuoteHelper(
-              ...quoteHelperArgs,
-            )
-            quotes.push({
-              ...quote,
-              swapperName: SwapperName.CowSwap,
-            })
-          } catch (error) {
-            console.error(error)
-          }
+        const swappers = [
+          {
+            isEnabled: OsmosisSwap,
+            swapperName: SwapperName.Osmosis,
+            getTradeQuote: getOsmosisTradeQuoteHelper,
+          },
+          {
+            isEnabled: LifiSwap,
+            swapperName: SwapperName.LIFI,
+            getTradeQuote: getLifiTradeQuoteHelper,
+          },
+          {
+            isEnabled: ThorSwap,
+            swapperName: SwapperName.Thorchain,
+            getTradeQuote: getThorTradeQuoteHelper,
+          },
+          {
+            isEnabled: ZrxSwap,
+            swapperName: SwapperName.Zrx,
+            getTradeQuote: getZrxTradeQuoteHelper,
+          },
+          {
+            isEnabled: OneInch,
+            swapperName: SwapperName.OneInch,
+            getTradeQuote: getOneInchTradeQuoteHelper,
+          },
+          {
+            isEnabled: Cowswap,
+            swapperName: SwapperName.CowSwap,
+            getTradeQuote: getCowTradeQuoteHelper,
+          },
+        ]
+
+        const quotes = await Promise.all(
+          swappers
+            .filter(({ isEnabled }) => isEnabled)
+            .map(({ swapperName, getTradeQuote }) =>
+              timeout<TradeQuote2, SwapErrorRight>(
+                getTradeQuote(getTradeQuoteInput, state),
+                TIMEOUT_ERROR,
+              ).then(quote => ({
+                ...quote,
+                swapperName,
```

Comment:
Definitely not for this PR but this broke my brain, I was expecting this to be an akschual quote not a result, if we want to make things cleaner in the future we could give a better home to the quote in a `{quoteResult: quote, swapperName}` resolution, but very very nitpicky, spreading the result object and adding a property to it is perfectly fine